### PR TITLE
Add gzip support

### PR DESCRIPTION
**Pull request by nhi-nguyen**

### CODE REVIEW

File: analytics/request.py [hunk 1]
```diff
@@ -2,29 +2,40 @@
 from dateutil.tz import tzutc
 import logging
 import json
-from analytics.version import VERSION
```
Comment (f2prateek) [inline analytics/request.py]: I think we still need this import?

File: analytics/request.py [hunk 2]
```diff
@@ -2,29 +2,40 @@
...
-def post(write_key, host=None, **kwargs):
+def post(write_key, host=None, gzip=False, **kwargs):
     """Post the `kwargs` to the API"""
     log = logging.getLogger('segment')
     body = kwargs
     body["sentAt"] = datetime.utcnow().replace(tzinfo=tzutc()).isoformat()
     url = remove_trailing_slash(host or 'https://api.segment.io') + '/v1/batch'
     auth = HTTPBasicAuth(write_key, '')
     data = json.dumps(body, cls=DatetimeSerializer)
+    log.debug('making request: %s', data)
```
Comment (f2prateek) [inline analytics/request.py]: was this left from debugging?

Comment (nhi-nguyen) [reply analytics/request.py]: this was also just moved up from a few lines below, to work with both compressed and uncompressed data.

File: analytics/request.py (hunk 1 above)
Comment (nhi-nguyen) [reply analytics/request.py]: I only moved it a few lines down :) (see below). This is for PEP8 conformance (import system modules before custom modules)

File: analytics/request.py (hunk 2 above)
Comment (f2prateek) [reply analytics/request.py]: gotcha!

**Pull request merged**